### PR TITLE
Show proper growl when websocket connection failed.

### PR DESCRIPTION
**Pull request by loganhz**
[rancher/rancher#7389](https://github.com/rancher/rancher/issues/7389)

### CODE REVIEW

File: app/mixins/subscribe.js [hunk 1]
```diff
@@ -160,6 +165,17 @@ export default Ember.Mixin.create({
...
   subscribeConnected: function(tries,msec) {
     this.set('connected', true);
+    this.set('wasConnected', true);
+
+    if( this.get('warningShown') ) {
+      this.get('growl').close();
+      this.set('warningShown', false);
+    }
+
+    if ( this.get('disconnectedTimer') )
```
Comment (vincent99) [inline app/mixins/subscribe.js]: Cancel does nothing if given null/an invalid timer, so the `if()` wrapper isn't needed

File: app/utils/socket.js [hunk 2]
```diff
@@ -209,15 +209,20 @@ export default Ember.Object.extend(Ember.Evented, {
...
+
+      const intl = window.l('service:intl');
+      let warningMessage = intl.t('growl.webSocket.connecting.proxyIssueWarning');
+      if ( isSafari && this.get('url').indexOf('wss://') === 0 )
       {
-        safariWarningShown = true;
-        window.l('service:growl').error('Error connecting to WebSocket','Safari does not support WebSockets connecting to a self-signed or unrecognized certificate.  Use http:// instead of https:// or reconfigure the server with a valid certificate from a recognized certificate authority.  Streaming stats, logs, shell/console, and auto-updating of the state of resources in the UI will not work until this is resolved.');
+        warningMessage += intl.t('growl.webSocket.connecting.safariCertWarning');
       }
+      warningMessage += intl.t('growl.webSocket.connecting.warning');
```
Comment (vincent99) [inline app/utils/socket.js]: Combine proxyissuewarning and warning (see translation file below) with the optional safari message below it

File: app/utils/socket.js [hunk 3]
```diff
@@ -209,15 +209,20 @@ export default Ember.Object.extend(Ember.Evented, {
...
       this.set('_state', DISCONNECTED);
-      if ( !safariWarningShown )
+
+      const intl = window.l('service:intl');
+      let warningMessage = intl.t('growl.webSocket.connecting.proxyIssueWarning');
+      if ( isSafari && this.get('url').indexOf('wss://') === 0 )
       {
-        safariWarningShown = true;
-        window.l('service:growl').error('Error connecting to WebSocket','Safari does not support WebSockets connecting to a self-signed or unrecognized certificate.  Use http:// instead of https:// or reconfigure the server with a valid certificate from a recognized certificate authority.  Streaming stats, logs, shell/console, and auto-updating of the state of resources in the UI will not work until this is resolved.');
+        warningMessage += intl.t('growl.webSocket.connecting.safariCertWarning');
```
Comment (vincent99) [inline app/utils/socket.js]: Add a newline or two to separate the regular message from safari one

File: app/mixins/subscribe.js [hunk 4]
```diff
@@ -189,6 +205,24 @@ export default Ember.Mixin.create({
     console.log('Subscribe disconnected ' + this.forStr());
     if ( this.get('reconnect') ) {
       this.connectSubscribe();
+      this.showDisconnectedWarning();
+    }
+  },
+
+  showDisconnectedWarning: function() {
+    if( !this.get('warningShown') && this.get('wasConnected') ) {
```
[the remark this answers is not in the document]
Comment (loganhz) [reply app/mixins/subscribe.js]: there is `this.set('warningShown', false);` in `subscribeConnected `. So I think it is ok?

[the remark this answers is not in the document]
Comment (vincent99) [reply app/mixins/subscribe.js]: Yes, I missed it

**Pull request merged**